src/coap_blockwise.c

```diff
@@ -80,24 +81,27 @@ static void on_block_sent(struct golioth_client *client,
 
 // Function to call the application's read block callback after a successful
 // blockwise upload
-static void call_read_block_callback(struct blockwise_transfer *ctx)
+static int call_read_block_callback(struct blockwise_transfer *ctx)
 {
+    size_t block_size = ctx->block_size;
     int err = ctx->callback.read_cb(ctx->block_idx,
                                     ctx->block_buffer,
-                                    &ctx->block_size,
+                                    &block_size,
                                     &ctx->is_last,
                                     ctx->callback_arg);
     if (err != GOLIOTH_OK)
     {
-        // TODO: handle application callback error
+        return -1 * err;
```

Comment:
why not `return -err;`?

Reply:
Honestly I was just working on this too late last night